### PR TITLE
simplifies logging

### PR DESCRIPTION
**Pull request by blundell**
Removes NoTils Logger in favour of one that doesn't query the stacktraces when it logs.

This won't impact performance in production but will contribute to perceived performance in debugging.

The test here was using the advanced demo and pressing "download a batch of items" the screenies show from start to completion.

| before |
| --- |
|screen shot 2015-07-24 at 15 23 42|

| after |
| --- |
|screen shot 2015-07-24 at 15 21 58|

LLog == Library Log

screen shot 2015-07-24 at 15 36 07

### CODE REVIEW

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/186/
Test FAILed.

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/187/
Test FAILed.

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/188/
Test FAILed.

File: library/src/main/java/com/novoda/downloadmanager/lib/GlobalState.java [hunk 1]
```diff
@@ -20,6 +22,7 @@ public static boolean hasVerboseLogging() {
     }
 
     public static void setVerboseLogging(boolean verboseLogging) {
+        LLog.setShowLogs(verboseLogging);
```
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/GlobalState.java]: this is now how logging is determined to be on or off. It is off by default for builds. Can be turned on through the builder when used by clients.

File: demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/delete/DeleteActivity.java [hunk 2]
```diff
@@ -39,14 +39,15 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        deleteAdapter = new DeleteAdapter(new ArrayList<Download>(), new DeleteAdapter.Listener() {
+        deleteAdapter = new DeleteAdapter(
+                new ArrayList<Download>(), new DeleteAdapter.Listener() {
             @Override
             public void onDelete(Download download) {
                 downloadManager.removeBatches(download.getBatchId());
             }
-        });
+        }
+        );
```
Comment (zegnus) [inline demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/delete/DeleteActivity.java]: Ups this is at the same level?

File: library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
+package com.novoda.downloadmanager.lib.logger;
+
+/**
+ * Wrapper around Android LLog that can be easily toggled for libraries
+ */
+public final class LLog {
+
+    private static boolean INITIALISED = false;
+
+    public static String TAG = "DownloadManager";
```
Comment (zegnus) [inline library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java]: final?

File: demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/delete/DeleteActivity.java (hunk 2 above)
Comment (blundell) [inline demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/delete/DeleteActivity.java]: ooh interesting

Comment (blundell) [inline demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/delete/DeleteActivity.java]: fixed here https://github.com/novoda/download-manager/commit/341f3dcf9512442accd8153b8ee085e90e3c46eb

File: library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java (hunk 3 above)
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java]: yes mr findbugs :)

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/189/
Test PASSed.

File: library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java (hunk 3 above)
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java]: fixed here https://github.com/novoda/download-manager/commit/9f5bd7765ac25afc1284636fc1bc999be6cc48f9

File: library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    public static void wtf(Throwable t, Object... msg) {
+        if (shouldShowLogs()) {
+            android.util.Log.wtf(TAG, formatString(msg), t);
+        }
+    }
+
+    private static String formatString(Object... msg) {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (Object o : msg) {
+            String separator = " ";
```
Comment (zegnus) [inline library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java]: Outside the loop or just inject it?

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/190/
Test PASSed.

File: library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java (hunk 4 above)
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java]: ah i was let down by intellij's auto refactor, fixed here https://github.com/novoda/download-manager/commit/30e67cea66bf0a11153cb7ede6da95a1d72ed3db

**Comment by zegnus** [comment]
Now we have a log but we don't have the source code line anymore right?

**Comment by ouchadam** [comment]
that allocation diff is crazy

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/191/
Test PASSed.

**Comment by zegnus** [comment]
:+1: This is really helpful because we won't be misled by the log object creation, nice!

File: demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/extra_data/ExtraDataActivity.java [hunk 5]
```diff
@@ -48,7 +47,8 @@ protected void onCreate(Bundle savedInstanceState) {
                     public void onClick(@NonNull View v) {
                         enqueueSingleDownload();
                     }
-                });
```
Comment (ouchadam) [inline demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/extra_data/ExtraDataActivity.java]: formatting rampage

File: library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
+package com.novoda.downloadmanager.lib.logger;
+
+/**
+ * Wrapper around Android LLog that can be easily toggled for libraries
+ */
+public final class LLog {
```
Comment (ouchadam) [inline library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java]: `LLog?` maybe we should have another name because it'll be inherited

**Comment by blundell** [comment]
@zegnus right no more line numbers :(

File: library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java (hunk 6 above)
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java]: shorthand for `LibraryLog` nobody wants to write that. What do you propose?

Comment (ouchadam) [inline library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java]: not sure of a name at the moment, not a blocker for me though :+1:

**Comment by ouchadam** [comment]
LGTM :100:

**Comment by zegnus** [comment]
Could it be useful to have a Log version with the source code line number as an option?

**Comment by blundell** [comment]
@zegnus yes potential future option is to make them toggleable

**Comment by zegnus** [comment]
:+1: LGTM then

**Comment by amlcurran** [comment]
@blundell how comes the allocated memory goes up?

**Comment by ouchadam** [comment]
different state, the allocations don't move though!

**Comment by blundell** [comment]
@amlcurran think it's just flux of the state of the phone like Adam says, it wasn't the most scientific benchmark but the graph shows the general idea

File: library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    public static void wtf(Throwable t, Object... msg) {
+        if (shouldShowLogs()) {
+            android.util.Log.wtf(TAG, formatString(msg), t);
+        }
+    }
+
+    private static String formatString(Object... msg) {
+        StringBuilder stringBuilder = new StringBuilder();
```
Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java]: if we really wanted to have simpler Logger we could get rid of the varargs and just allow a string, that way we wouldn't need a StringBuilder every time

Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java]: and we're always creating a StringBuilder even if you only pass 1 string to this method

File: library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java (hunk 6 above)
Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/logger/LLog.java]: lightweight log? :tongue:

**Pull request merged**